datafusion/core/tests/sql/sql_api.rs

```diff
@@ -58,6 +58,19 @@ async fn unsupported_dml_returns_error() {
     ctx.sql_with_options(sql, options).await.unwrap();
 }
 
+#[tokio::test]
```

Comment:
its better to have this test in one of `.slt` files which runs end to end tests

Reply:
I suspect this might not be caught in a `.slt` test as this issue is not visible in the pretty-printed logical plan, and doesn't affect the physical plan or execution at all, afaict.

Reply:
Actually I see some slt tests failing on CI, I should look into that.

Reply:
Curiously, this was in fact tested in many sqllogictests. Because the tests do in fact cover checking that the physical output schema matches the logical output schema, which is pretty cool.

Let me know if you want to keep this Rust test as well, or if you rather that I remove it. It does test more behaviour than what the slts can test.

Reply:
I think keeping the rust level test is a good idea as it makes explicit the expected output schema